### PR TITLE
Improve coverage

### PR DESCRIPTION
**Pull request by tomschr**
This PR focus on improving coverage for `git.py`, `env.py`, and `lifecycle.py`. Other files aren't affected yet. It doesn't change functionality.

Coverage report raised from 96.8% (main) by 1% to 97.8%. :tada: Every percent counts. :laughing:

### CODE REVIEW

File: tests/utils/test_git.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import asyncio
 from pathlib import Path
 from subprocess import CompletedProcess
-from unittest.mock import AsyncMock
+from unittest.mock import AsyncMock, patch
```
Comment (sushant-suse) [inline tests/utils/test_git.py]: Hi Toms, this is a great addition to the test suite! Consolidating the `mock_execute_git` fixture makes the tests much more readable and easier to maintain.

I specifically like the comprehensive coverage for `ls_tree`. It hits all the key edge cases (empty output, missing repo, and command failures) that are critical for the reliability of the build process.

Small Suggestion: In `test_ls_tree_no_repo` and `test_ls_tree_command_failure`, you are using `unittest.mock.patch` for the logger. While this works perfectly (IMHO), consider using the `caplog` fixture in the future for a more `'pytest-native'` way of asserting on logs. No need to change it here though, as the current implementation is solid.

**Comment by sushant-suse** [comment]
> This PR focus on improving coverage for `git.py`, `env.py`, and `lifecycle.py`. Other files aren't affected yet. It doesn't change functionality.
> 
> Coverage report raised from 96.8% (main) by 1% to 97.8%. 🎉 Every percent counts. 😆

Hi @tomschr, definitely every percent counts😆

Great work on the coverage improvements. I especially appreciate the cleanup of the `mock_execute_git` fixture as it makes the utility tests much more maintainable. Approved.
